### PR TITLE
Remove v2 http proxy 

### PR DESCRIPTION
**Pull request by ahrtr**
Both v2 server and v2 commands has already been removed in main, so there is no reason to keep v2 http proxy. 

If an etcd server is already running as a v2 proxy, then it can't be upgraded to 3.6.

### CODE REVIEW

**Comment by ahrtr** [comment]
@ptabor @serathius @spzala Both v2 server and v2 commands have already been removed in main branch, I don't see any reason to keep the v2 http proxy.  So this PR is going to remove v2 http proxy. 

If a user keeps etcd cluster to 3.5 or older versions, while upgrades the v2 proxy to 3.6, then they will run into  issue. But it doesn't make sense for a user to do it. On the other hand, grpc proxy has already been there for years. So I think it's safe to remove v2 http proxy. 

WDYT?

**Review by serathius — APPROVED** [review]
cc @ptabor @spzala

File: server/etcdmain/help.go [hunk 1]
```diff
@@ -132,9 +133,8 @@ Clustering:
     V3 discovery: username[:password] for authentication (prompt if password is not supplied).
   --discovery-password ''
     V3 discovery: password for authentication (if this option is used, --user option shouldn't include password).
-  --discovery-fallback 'proxy'
-    Expected behavior ('exit' or 'proxy') when discovery services fails.
-    "proxy" supports v2 API only.
+  --discovery-fallback 'exit'
+    Expected behavior ('exit') when discovery services fails. Note that v2 proxy is deprecated.
```
[the remark this answers is not in the document]
Comment (ahrtr) [reply server/etcdmain/help.go]: Updated, thx

**Comment by ahrtr** [comment]
With the approval of this PR, actually the other PR [14032](https://github.com/etcd-io/etcd/pull/14032) isn't needed any more. But there were already lots of discussion/comments on that PR before I realized that we need to completely remove v2 http proxy. 

So I'd like honor the contributor's effort, please try to approve and merge that PR [14032](https://github.com/etcd-io/etcd/pull/14032) firstly, and I am happy to rebase this one. It's just a simple & minor change, and it should be safe to merge that PR. 

cc @spzala @serathius @ptabor

**Review by ptabor — APPROVED** [review]

**Pull request merged**